src/linear_solver_routines/block_col_projection.jl

```diff
@@ -0,0 +1,52 @@
+# This file is pat of RLinearAlgebra.jl
+# This was written by Nathaniel Pritchard
+"""
+    LinSysBlkColProj <: LinSysBlkColProjection
+
+A mutable structure that represents a standard block column projection method.
+
+# Aliases
+- `BlockCoordinateDescent`
+
+# Fields
+- `alpha::Float64`, a relaxation parameter that should be set between `0.0` and `2.0`
+- `G::GentData`, a buffer to store important information related to the Gentleman's 
+incremental QR least squares solver.
+
+Calling the constructor `LinSysBlkColProj()` defaults the relaxation parameter to `1.0`.
+
+"""
+mutable struct LinSysBlkColProj <: LinSysBlkColProjection 
```

Comment:
I think rather it would be better to allow people to specify the inner solver data structure itself. Then instead of having to create a newly named block projection method for each inner solver, which will lead to a lot of rewriting the same code, we can instead just need to implement an instantiate_inner_solver and ldiv! function for each inner solver.

Reply:
Is this not what the structures allow for? For instance, you would not want to use CG for a Block Column Projection as you would not have symmetry. So this would preclude this subsolver from being used. Also, a subsolver can be the child of the union of two types, so it would not have to be duplicated. Maybe I am misunderstanding your comment here.

Reply:
For the time being the names have been changed to indicate what solver is being called.